File: go/isogram/isogram.go
// Package isogram provides helpful method to handle with isogram sentences
// An isogram (also known as a "nonpattern word") is a word or phrase without a repeating letter, however spaces and
// hyphens are allowed to appear multiple times
package isogram

import (
	"unicode"
)

// IsIsogram determine if a word or phrase is an isogram
func IsIsogram(phrase string) bool {
	letters := make(map[rune]bool)

	for _, letter := range phrase {
		if letter == ' ' || letter == '-' {
			continue
		}

		normalizedLetter := unicode.ToUpper(letter)
		if letters[normalizedLetter] {
			return false
		}
		letters[normalizedLetter] = true
	}
	return true
}

File: javascript/isogram/isogram.js
export const isIsogram = (sentence) => {
  const letters = {}

  return sentence
    .toUpperCase()
    .split('')
    .every(letter => {
      if (letter !== ' ' && letter !== '-' && letters[letter]) {
        return false
      }

      letters[letter] = true
      return true
    })
};
